test/block.spec.ts

```diff
@@ -129,4 +130,109 @@ tape('[Block]: block functions', function(t) {
     st.equal(typeof block.toJSON(true), 'object')
     st.end()
   })
+
+  t.test('should validate a valid block', async st => {
+    const [blockchain, lastBlock] = await setupBlockchain()
+    const nextBlock = new Block(
+      {
+        header: {
+          number: 2,
+          parentHash: lastBlock.header.hash(),
+          gasLimit: lastBlock.header.gasLimit,
+          timestamp: Date.now(),
+          extraData: Buffer.from('abc'),
+        },
+      },
+      { chain: 'ropsten' },
+    )
+    nextBlock.header.difficulty = toBuffer(nextBlock.header.canonicalDifficulty(lastBlock))
+    try {
+      await nextBlock.validate(blockchain)
+      st.ok(true)
+      st.end()
+    } catch (error) {
+      st.fail(error)
+      st.end()
+    }
+  })
+
+  t.test('should throw on invalid uncle hash', async st => {
+    const [blockchain, lastBlock] = await setupBlockchain()
+    const nextBlock = new Block(
+      {
+        header: {
+          number: 2,
+          parentHash: lastBlock.header.hash(),
+          gasLimit: lastBlock.header.gasLimit,
+          timestamp: Date.now(),
+          extraData: Buffer.from('abc'),
+          uncleHash: Buffer.from('testFail'),
+        },
+      },
+      { chain: 'ropsten' },
+    )
+    nextBlock.header.difficulty = toBuffer(nextBlock.header.canonicalDifficulty(lastBlock))
+    try {
+      await nextBlock.validate(blockchain)
+      st.fail('should not validate')
+      st.end()
+    } catch (error) {
+      st.ok(true)
+      st.end()
+    }
+  })
```

Comment:
Would eventually be nice on such cases to more explicitly describe what makes the hash invalid to make this more transparent for follow-up devs, but otherwise ok.